auth/api/v1/api.go

```diff
@@ -276,7 +276,46 @@ func (w Wrapper) CreateJwtGrant(ctx echo.Context) error {
 		return core.InvalidInputError(err.Error())
 	}
 
-	return ctx.JSON(http.StatusOK, JwtBearerTokenResponse{BearerToken: response.BearerToken})
+	return ctx.JSON(http.StatusOK, JwtGrantResponse{BearerToken: response.BearerToken})
+}
+
+// RequestToken handles the http request (from the vendor's EPD/XIS) for creating a JWT grant and using it as authorization grant to get an access token from the remote Nuts node.
+func (w Wrapper) RequestToken(ctx echo.Context) error {
+	requestBody := &RequestTokenRequest{}
+	if err := ctx.Bind(requestBody); err != nil {
+		return err
+	}
+
+	request := services.CreateJwtGrantRequest{
+		Actor:         requestBody.Actor,
+		Custodian:     requestBody.Custodian,
+		IdentityToken: &requestBody.Identity,
+		Service:       requestBody.Service,
+		Subject:       requestBody.Subject,
+	}
+	jwtGrantResponse, err := w.Auth.OAuthClient().CreateJwtGrant(request)
+	if err != nil {
+		return core.InvalidInputError(err.Error())
+	}
+
+	custodianDID, err := did.ParseDID(requestBody.Custodian)
+	if err != nil {
+		return core.InvalidInputError(err.Error())
+	}
+
+	endpointURL, err := w.Auth.OAuthClient().GetOAuthEndpointURL(requestBody.Service, *custodianDID)
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, fmt.Errorf("unable to find the oauth2 service endpoint of the custodian: %w", err))
+	}
+
+	client := HTTPClient{Timeout: 30 * time.Second}
```

Comment:
I agree. Added that in the last commit. Since `HTTPClient` is not a "service" I'm unable to pass the configuration parameter properly so for now it's embedded in the auth services as `HTTPTimeout() time.Duration`. It feels a bit odd so if you have another suggestion please let me know!